### PR TITLE
[EventDispatcher] Deprecate configuring tags on RegisterListenersPass

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Follow-up to #40468.

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php [hunk 1]
```diff
@@ -50,21 +50,29 @@ public function __construct(string $dispatcherService = 'event_dispatcher', stri
     /**
      * @return $this
      */
-    public function setHotPathEvents(array $hotPathEvents, string $tagName = 'container.hot_path')
+    public function setHotPathEvents(array $hotPathEvents)
```
Comment (chalasr) [inline src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php:53]: BC break?

Comment (stof) [reply src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php:53]: As the class is not final, indeed

Comment (stof) [reply src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php:53]: hmm, actually no. Child classes can add optional arguments, so this is indeed not a BC break.

Comment (chalasr) [reply src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php:53]: right

**Review by chalasr — APPROVED** [review]

**Pull request merged**